window_manager/src/MsWindow.hpp

```diff
@@ -0,0 +1,135 @@
+#ifndef RDMN_MS_WINDOW_HPP
+#define RDMN_MS_WINDOW_HPP
+
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+
+#include <atomic>
+#include <future>
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <type_traits>
+
+#include <rcbe-engine/datamodel/rendering/RenderingContext.hpp>
+#include <rcbe-engine/datamodel/system/WindowContext.hpp>
+#include <rcbe-engine/datamodel/system/window_config.hpp>
+#include <rcbe-engine/core/AbstractInputManager.hpp>
+#include <rcbe-engine/core/windows_input_event.hpp>
+#include <rcbe-engine/renderer/Renderer.hpp>
+
+#include <rdmn-engine/logger/trivial_logger.hpp>
+
+namespace rcbe::core {
+template <typename Implementation>
+class BaseWindow {
+public:
+	static LRESULT CALLBACK windowLoopHandler(HWND hwnd, UINT msg, WPARAM wparam, LPARAM lparam) {
+		Implementation* impl_this = nullptr;
```

Comment:
Well I guess one window per process.

Reply:
Nope, all good. Register pointer in system, then can operate on different windows.